format-common/src/main/java/io/cdap/plugin/format/input/PathTrackingInputFormatProvider.java

```diff
@@ -63,6 +65,18 @@ protected void validate() {
     // no-op
   }
 
+  protected Schema getSchema(Schema dataSchema) {
+    List<Schema.Field> fields = new ArrayList<>(dataSchema.getFields());
+    addPathField(fields);
+    return Schema.recordOf(dataSchema.getRecordName(), fields);
+  }
+
+  private void addPathField(List<Schema.Field> dataFields) {
```

Comment:
make this protected. Both these methods should also take in a FailureCollector and add a failure on the path field if the field already exists in the data fields.

Reply:
should also refactor the blob and text formats to call these methods instead of their own logic for path field.

Reply:
Changed to protected.Added failureCollector if pathfield value already exists.Will refactor blob,text formats separately